apps/api/e2e/utils/api-types.ts

```diff
@@ -682,3 +687,65 @@ export interface GlobalLeaderboardResponse extends ApiResponse {
     hasMore: boolean;
   };
 }
+
+// ===========================
+// Vote-related types
+// ===========================
+
+/**
+ * Response type for casting a vote
+ */
+export interface VoteResponse extends ApiResponse {
+  vote: {
+    id: string;
+    userId: string;
+    agentId: string;
+    competitionId: string;
+    createdAt: string;
+  };
+}
+
+/**
+ * Response type for getting user votes
+ */
+export interface UserVotesResponse extends ApiResponse {
+  votes: Array<{
+    id: string;
+    agentId: string;
+    competitionId: string;
+    createdAt: string;
+  }>;
+  pagination?: {
+    total: number;
+    limit: number;
+    offset: number;
+    hasMore: boolean;
+  };
+}
+
+/**
+ * User vote info object
+ */
+export interface UserVoteInfo {
+  hasVoted: boolean;
+  votedAgentId?: string;
+  votedAt?: string;
+}
+
+/**
+ * Competition voting state object
+ */
+export interface CompetitionVotingState {
+  canVote: boolean;
+  reason?: string;
+  userVoteInfo: UserVoteInfo;
```

Comment:
it feels like `status` is the right word instead of `state`? i dont have a strong preference tho, if you feel strongly otherwise.
- `CompetitionVotingState` -> `CompetitionVotingStatus`

and then we can also just keep this a little simpler imo:
`userVoteInfo` -> `info`